source/MaterialXRenderGlsl/GlslValidator.cpp

```diff
@@ -313,6 +315,66 @@ void GlslValidator::validateCreation(const StageMap& stages)
     _program->build();
 }
 
+void GlslValidator::renderScreenSpaceQuad(GenContext& context)
+{
+    bindTarget(true);
+    _type = context.getOptions().textureSpaceInputType;
+    (_type == "color3" || _type == "color4")? glEnable(GL_FRAMEBUFFER_SRGB) : glDisable(GL_FRAMEBUFFER_SRGB);
```

Comment:
I'm not sure I understand why a color3 vs a color4 determines whether to perform gamma correction?
Perhaps this is for normal map generation. In which case I would allow a user setting to turn this on or off then no knowledge of this textureSpaceInputType is required.